### PR TITLE
FEATURE: Add opt-in screenshots to the web app manifest

### PR DESCRIPTION
**Pull request by xfalcox**

### CODE REVIEW

File: app/controllers/metadata_controller.rb [hunk 1]
```diff
@@ -124,6 +124,21 @@ def default_manifest
       manifest[:icons] << icon_entry
     end
 
+    SiteSetting.manifest_screenshots.split('|').each do |image|
+      next unless Discourse.store.has_been_uploaded?(image)
+
+      upload = Upload.find_by(url: image.delete_prefix(Discourse.base_url))
```
Comment (davidtaylorhq) [inline app/controllers/metadata_controller.rb:130]: Does this work for S3 uploads?

Comment (xfalcox) [reply app/controllers/metadata_controller.rb:130]: Oh yeah, that will fail.

I tried to find a method to go from Upload full URL to Upload object but did not find a ready made method for that, are you aware of any?

Comment (davidtaylorhq) [reply app/controllers/metadata_controller.rb:130]: Maybe extract the SHA and then lookup based on that?

https://github.com/discourse/discourse/blob/a96a5db0fb516c3b88b225827eda47af63995d64/app/models/post.rb#L1013-L1018

Comment (xfalcox) [reply app/controllers/metadata_controller.rb]: Should work now @davidtaylorhq !

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**